loadtest/control/simulcontroller/actions.go

```diff
@@ -1831,3 +1832,156 @@ func (c *SimulController) createPostReminder(u user.User) control.UserActionResp
 
 	return control.UserActionResponse{Info: fmt.Sprintf("created post reminder, id %s", post.Id)}
 }
+
+func (c *SimulController) clickUserProfile(u user.User) control.UserActionResponse {
+	ch, err := u.Store().CurrentChannel()
+	if errors.Is(err, memstore.ErrChannelNotFound) {
+		return control.UserActionResponse{Info: "current channel is not set"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	post, err := u.Store().RandomPostForChannel(ch.Id)
+	if errors.Is(err, memstore.ErrPostNotFound) {
+		return control.UserActionResponse{Info: fmt.Sprintf("no post in channel: %s", ch.Id)}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	// We simulate a click on the user profile who wrote the post.
+	// The webapp will also do an additional viewChannel request once per-channel.
+	// But we avoid that for now because viewChannel is already handled separately.
+	if err := u.GetChannelMember(ch.Id, post.UserId); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	// If it's a DM/GM channel, the webapp still sends the current team
+	// the user is part of
+	if ch.TeamId == "" {
+		team, err := u.Store().CurrentTeam()
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		if team != nil {
+			ch.TeamId = team.Id
+		}
```

Comment:
Should we perhaps warn or error if the current team is missing?

Reply:
done